nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/test/java/org/apache/nifi/processors/influxdb/AbstractITInfluxDB.java

```diff
@@ -0,0 +1,79 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.influxdb;
+import org.apache.nifi.util.TestRunner;
+import org.influxdb.InfluxDB;
+import org.influxdb.InfluxDBFactory;
+import org.influxdb.dto.Query;
+import org.influxdb.dto.QueryResult;
+import org.junit.After;
+
+/**
+ * Base integration test class for InfluxDB processors
+ */
+public class AbstractITInfluxDB {
+    protected TestRunner runner;
+    protected InfluxDB influxDB;
+    protected String dbName = "test";
+    protected String dbUrl = "http://localhost:8086";
+    protected String user = "admin";
+    protected String password = "admin";
+    protected static final String DEFAULT_RETENTION_POLICY = "autogen";
+
+    protected void initInfluxDB() throws InterruptedException, Exception {
+        influxDB = InfluxDBFactory.connect(dbUrl,user,password);
+        if ( influxDB.databaseExists(dbName) ) {
```

Comment:
I would strongly suggest moving this to the `@After` section because it'll make everything behave in one clean arc of setup -> test -> cleanup.

Reply:
I've refactored the code and call it both from `Before` and `After` to make sure that the test setup and tear down is clean.